src/services/dataRenderService.ts

```diff
@@ -332,48 +332,39 @@ export class DataRenderService {
 
             this.applyOnObjectStylesToCircleText(text);
 
-        } else {            
+        } else {
             element.selectAll(DataRenderService.CircleText.selectorName).remove();
         }
     }
 
     private applyOnObjectStylesToCircleText(text: d3Selection<SVGTextElement, number, SVGGElement, number>): void{
         text
-        .attr(SubSelectableObjectNameAttribute, AsterPlotObjectNames.Ticks.name)
-        .attr(SubSelectableDisplayNameAttribute, this.localizationManager.getDisplayName(AsterPlotObjectNames.Ticks.displayNameKey))
-        .attr(SubSelectableTypeAttribute, SubSelectionStylesType.Text)
-        .classed(HtmlSubSelectableClass, this.formatMode && this.settings.outerLine.showGridTicksValues.value);
+            .attr(SubSelectableObjectNameAttribute, AsterPlotObjectNames.Ticks.name)
+            .attr(SubSelectableDisplayNameAttribute, this.localizationManager.getDisplayName(AsterPlotObjectNames.Ticks.displayNameKey))
+            .attr(SubSelectableTypeAttribute, SubSelectionStylesType.Text)
+            .classed(HtmlSubSelectableClass, this.formatMode && this.settings.outerLine.showGridTicksValues.value);
     }
 
-    private drawArcCircles(
+    private drawInnerAndOuterCircles(
         element: d3Selection<SVGGElement, null, HTMLElement, null>,
         circleClassName: ClassAndSelector,
-        radius: number,
+        radius: number
     ): void {
-        const selection = element.selectAll<SVGPathElement, d3PieArcDatum<AsterDataPoint>>(circleClassName.selectorName).data([this.dataPoints[0]]);
+        const selection = element.selectAll<SVGCircleElement, d3PieArcDatum<AsterDataPoint>>(circleClassName.selectorName).data([this.dataPoints[0]]);
```

Comment:
No need to pass datapoints to data() now. You can pass [radius] only. That will make code more clear, we don't need to depend on datapoints for these circles, also it might slightly improve performance.